salt/modules/dockermod.py

```diff
@@ -910,8 +911,12 @@ def compare_container(first, second, ignore=None):
                     ret.setdefault(conf_dict, {})[item] = {'old': image1, 'new': image2}
             else:
                 if item == 'Links':
-                    val1 = sorted(_scrub_links(val1, first))
-                    val2 = sorted(_scrub_links(val2, second))
+                    val1 = _scrub_links(val1, first)
+                    val2 = _scrub_links(val2, second)
+                if isinstance(val1, collections.Iterable):
```

Comment:
Looks like some legit failed tests there, this isn't appropriate for all `Config` and `HostConfig` values, for example, `Config` `Cmd`.